docker-images/stunnel-zookeeper/scripts/stunnel_config_generator.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+	      
+CERTS=/etc/stunnel/certs/
+
+echo "foreground = yes"
+#echo "pid=/tmp/stunnel.pid"
+echo "debug = info"
+ 
+declare -a PORTS=("2888" "3888")
+for port in "${PORTS[@]}"
+do
+NODE=1
+ while [ $NODE -le $ZOOKEEPER_NODE_COUNT ]; do
+	        
+		PEER=${BASE_HOSTNAME}-$((NODE-1))
+		KEY_AND_CERT=${CERTS}${PEER}	
+		PEM=/tmp/${PEER}.pem
+
+		# Not necessarily the way to create pem file, might have to use openssl
+		cat ${KEY_AND_CERT}.crt ${KEY_AND_CERT}.key > ${PEM}
+		chmod 640 ${PEM}
+
+		# Stunnel client configuration
+		cat <<-EOF
+		[${PEER})]
+		client=yes
+		cert=${PEM}
```

Comment:
I think that it should be the "internal-ca" certificate so you should use `CAfile` parameter and not `cert`.
In this client configuration, the stunnel is connecting to a remote peer which will send the server certificate signed by the "internal-ca" cert; for this reason the stunnel client has to verify the signed certificate (received by the peer during the TLS handshake) using such CA cert.